tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+                        "discriminator": 1234,
+                        "roles": [
+                            1234,
+                            5678
+                        ],
+                        "in_guild": False
+                    }
+                ],
+                "raise_error": True,
+                "payload": {
+                    "avatar_hash": getattr(self.user, "avatar", 0),
+                    "discriminator": int(getattr(self.user, "discriminator", 0)),
+                    "id": self.user.id,
+                    "in_guild": False,
+                    "name": getattr(self.user, "name", "Name unknown"),
+                    "roles": []
+                }
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["post_result"]
+            error = case["raise_error"]
+            payload = case["payload"]
+
+            with self.subTest(args=args, result=expected, error=error, payload=payload):
+                self.ctx.bot.api_client.post.return_value = expected
+
+                if error:
+                    self.ctx.bot.api_client.post.side_effect = ResponseCodeError(AsyncMock(), expected)
+
+                result = await post_user(*args)
+
+                if error:
+                    self.assertIsNone(result)
+                else:
+                    self.assertEqual(result, expected)
+
+                if not error:
+                    self.bot.api_client.post.assert_awaited_once_with("bot/users", json=payload)
+
+                self.bot.api_client.post.reset_mock(side_effect=True)
+
+    async def test_send_private_embed(self):
+        """Test does `send_private_embed` return correct value."""
+        test_cases = [
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": True,
+                "raised_exception": None
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": HTTPException(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": Forbidden(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": NotFound(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            raised: Union[Forbidden, HTTPException, NotFound, None] = case["raised_exception"]
+
+            with self.subTest(args=args, expected=expected, raised=raised):
+                if raised:
```

Comment:
There's no need for this conditional. You used a value of `None` in the test case when an exception shouldn't be raised. Well, if `side_effect = None` then that's the same thing as not having a side effect. Therefore, it's perfectly safe to assign `None` to the `side_effect` when an exception shouldn't be raised.